### PR TITLE
feat: Add callback functionality for capture events

### PR DESCRIPTION
**Pull request by bjerringgaard**

### CODE REVIEW

File: src/controllers/PaymentsCallbackController.php [hunk 1]
```diff
@@ -51,117 +55,77 @@ public function beforeAction($action): bool
...
 	public function actionContinue(string $transactionReference = null): Response
 	{
-		//Get transaction and order
-		$authTransaction = Plugin::getInstance()->transactions->getTransactionByHash($transactionReference);
-		$order = Plugin::getInstance()->orders->getOrderById($authTransaction->orderId);
-		//Order is already paid
-		if ($order->getIsPaid()) {
-			return Craft::$app->getResponse()->redirect($order->returnUrl);
+		// Check if transaction reference is set
+		//TODO: Update to use same logic as notify
```
Comment (NicolajKN) [inline src/controllers/PaymentsCallbackController.php:67]: Should this be addressed?

File: src/controllers/PaymentsCallbackController.php [hunk 2]
```diff
@@ -51,117 +55,77 @@ public function beforeAction($action): bool
...
 	 * @param string|null $transactionReference
 	 * @return Response
 	 */
 	public function actionCancel(string $transactionReference = null): Response
 	{
-		//Get transaction and order
-		$authTransaction = Plugin::getInstance()->transactions->getTransactionByHash($transactionReference);
-		$order = Plugin::getInstance()->orders->getOrderById($authTransaction->orderId);
+		// Check if transaction reference is set
+		//TODO: Update to use same logic as notify
```
Comment (NicolajKN) [inline src/controllers/PaymentsCallbackController.php:95]: Same?

File: src/Plugin.php [hunk 3]
```diff
@@ -130,6 +130,7 @@ function (RegisterUrlRulesEvent $event) {
 				$event->rules = array_merge($event->rules, [
 					'quickpay/callbacks/payments/continue/<transactionReference>' => 'commerce-quickpay/payments-callback/continue',
 					'quickpay/callbacks/payments/notify/<transactionReference>' => 'commerce-quickpay/payments-callback/notify',
```
Comment (NicolajKN) [inline src/Plugin.php]: Deprecate

File: src/services/Orders.php [hunk 4]
```diff
@@ -69,6 +71,30 @@ public function getSuccessfulTransactionForOrder(Order $order): Transaction|bool
...
 	}
 
+	/**
+	 * Update the order status
+	 *
+	 * @param Order $order
+	 * @param BaseGatewayInterface $gateway
+	 * @return void
+	 */
+	public function updateOrderStatus(Order $order, BaseGatewayInterface $gateway): void
```
Comment (NicolajKN) [inline src/services/Orders.php]: Should be more descriptive, eg. setAfterCaptureStatus

File: src/services/PaymentsCallbackService.php [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+          break;
+
+        case Data::STATE_REJECTED:
+          //? If the parent transaction is not successful (Initial / Redirect), and the update State is "Rejected" quickpay has rejected the transaction
+          Quickpay::getInstance()->getTransactionService()->createAuthorize($data, Transaction::STATUS_FAILED, 'Transaction rejected.');
+          break;
+      }
+    }
+
+    // Get the last transaction state
```
Comment (NicolajKN) [inline src/services/PaymentsCallbackService.php]: Comment

File: src/services/TransactionService.php [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+    }
+
+    return $transaction;
+  }
+
+  /**
+   * Create a child transaction of the type Capture
+   *
+   * @param mixed $data
+   * @param TransactionModel $parent
```
Comment (NicolajKN) [inline src/services/TransactionService.php]: Mismatched with function

File: src/services/TransactionService.php [hunk 7]
```diff
@@ -0,0 +1,222 @@
...
+class TransactionService extends Component
+{
+  //* Create
+
+
+  /**
+   * Create a child transaction of the type Authorize
+   *
+   * @param mixed $data
+   * @param TransactionModel $parent
```
Comment (NicolajKN) [inline src/services/TransactionService.php]: Mismatched

File: src/services/TransactionService.php [hunk 8]
```diff
@@ -0,0 +1,222 @@
...
+   * @param string $status
+   * @param string $message
+   * @return TransactionModel
+   */
+  public function createAuthorize(mixed $data, string $status, string $message = ''): TransactionModel
```
Comment (NicolajKN) [inline src/services/TransactionService.php]: Mixed should be response model (also other methods in this class) + $parentId as first param

**Review by NicolajKN — CHANGES_REQUESTED** [review]

File: src/controllers/PaymentsCallbackController.php (hunk 1 above)
Comment (bjerringgaard) [reply src/controllers/PaymentsCallbackController.php:67]: Will be addressed in a later version, as to not change to much critical logic

File: src/controllers/PaymentsCallbackController.php (hunk 2 above)
Comment (bjerringgaard) [reply src/controllers/PaymentsCallbackController.php:95]: Will be addressed in a later version, as to not change to much critical logic

**Review by NicolajKN — APPROVED** [review]
Looks good, let's test it 🎉

**Pull request merged**